src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -522,12 +548,17 @@ public Attachment GetAttachment(DocumentsOperationContext context, string docume
                         metadata.TryGet(Constants.Documents.Metadata.ChangeVector, out string exitingDocumentCv) &&
                         exitingDocumentCv == changeVector)
                     {
-                        return _documentsStorage.AttachmentsStorage.GetAttachment(context, documentId, name, AttachmentType.Document, null);
+                        return _documentsStorage.AttachmentsStorage.GetAttachment(context, documentId, name, AttachmentType.Document, null, withoutStream);
                     }
                 }
 
                 return null;
             }
+            // We were invoked just for the purpose of checking the existence of the attachment, no need to actually read the stream from disk only to verify its there.
+            if (withoutStream)
+            {
+                return VerifyAttachmentStreamExist(context, attachment.Base64Hash)?attachment:null;
```

Comment:
Why is it possible to have an attachment without the stream?

Reply:
BTW it isn't obvious for the caller of this method that `withoutStream` parameter changes the logic and calls  `VerifyAttachmentStreamExist` - it looks the storage should have a dedicated method for that check

Reply:
This is an optimization so we won't need to fetch the stream, it is invoked just to verify the existence of a stream without having to read the stream from the disk.

Reply:
i have not changed the logic of the method even by a bit, was just trying to save the actual read of the stream from the disk where only its existence i care about.
I see that effort is not been appreciated next time I'll just invoke code like /databases/*/attachments HEAD